vendor/github.com/vektah/gqlparser/v2/ast/decode.go

```diff
@@ -0,0 +1,198 @@
+package ast
```

Comment:
OK, I'm surprised that this doesn't blow up in CI 😅 I think it points to a bling spot in our testing.

Concretely, if someone were to import OPA as a library (opa-envoy-plugin, the playground, many tools and products), the vendored changes _from this repo_ wouldn't get picked up, but the go modules would be taken from what's referenced in `go.mod` and `go.sum`. The function we're adding here thus wouldn't be available and the project wouldn't build.

I think we're at a cross roads here:

#### Path 1: internalizing the code

- We'd copy the gqlparser code (all of it), and this addition into `internal/gqlparser`. 
- Create a README in the folder, explaining the What and the Why.
- We could set up some sort of notification system to get informed about upstream changes, so that we're not missing out, or drift away too much, from the third party library. It seems like it's still under development in terms of bug fixes we'd be interested in, so a simple copy-and-forget is not the right choice, I think.

We've done this for the jwt signing/verifying code, and we've drifted quite far there, losing track of upstream changes, and didn't include a README. [`internal/gojsonschema` is another, perhaps better example.](https://github.com/open-policy-agent/opa/tree/main/internal/gojsonschema)

#### Path 2: upstream the change

The additions here don't strike me as controversial --- perhaps you could open an issue (or straightaway a PR) with that change on the vektah/gqlparser repo?

Depending on the turnaround time, we could still go with _Path 1_ temporarily.

🤔 I'd try (2.) before going down (1.)... 🚶

Reply:
@srenatus PR has been submitted upstream to [`vektah/gqlparser` #232](https://github.com/vektah/gqlparser/pull/232). 

We'll see if they're interested, and if we haven't gotten a response in a week, I'd say we should then investigate option (1.) more seriously.

Reply:
Thanks! I'm still wondering about the correctness of that, though -- have you had a chance to consider https://github.com/open-policy-agent/opa/pull/4752#discussion_r930757316?

Reply:
I had not properly considered that case before the upstream merged the changes. :upside_down_face:  If needed, I'll see what sort of fix might be needed, and then will submit a patch upstream.